src/ui/components/NagDismiss.tsx

```diff
@@ -0,0 +1,19 @@
+import React, { ComponentType, useEffect } from "react";
+
+import { PropsFromRedux } from "devtools/client/debugger/src/components/PrimaryPanes/SourcesTree";
+import { useNag } from "replay-next/src/hooks/useNag";
+import { Nag } from "shared/graphql/types";
+
+const NagDismiss = <P extends PropsFromRedux>(WrappedComponent: ComponentType<P>, nagType: Nag) => {
+  return (props: P) => {
+    const [, dismissNag] = useNag(nagType);
+
+    useEffect(() => {
+      dismissNag();
+    }, [dismissNag]);
+
+    return <WrappedComponent {...props} />;
+  };
+};
+
+export default NagDismiss;
```

Comment:
I don't think this component is being used anywhere, so we should delete it.

_If_ we wanted to go with an auto-dismiss-nag-on-mount approach, I think we'd probably want to use a hook instead since it's a lot "lighter" in terms of the React overhead.

```ts
function useNagDismissOnMount(nagType: Nag) {
  const [, dismissNag] = useNag(nagType);

  useEffect(() => {
    dismissNag();
  }, [dismissNag]);
}
```